modules/hcp-topology-spread-constraint.adoc

```diff
@@ -0,0 +1,49 @@
+// Module included in the following assemblies:
+//
+// * hosted_control_planes/hcp-manage/hcp-manage-virt.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="hcp-topology-spread-constraint_{context}"]
+= Spreading node pool VMs by using topologySpreadConstraint
+
+By default, KubeVirt virtual machines (VMs) created by a node pool are scheduled to any available node with the appropriate capacity to run the VM. A `topologySpreadConstraint` constraint spreads VMs of a node pool on multiple nodes. For example, if you are performing a live migration, you use `topologySpreadConstraint` with the descheduler so that VMs are continuously distributed on multiple nodes rather than a single node.
```

Comment:
> For example, if you are performing a live migration, you use `topologySpreadConstraint` with the descheduler so that VMs are continuously distributed on multiple nodes rather than a single node.

The topologySpreadConstraint is always set by the hypershift controller when preparing the VMs for the node pool.
Nothing is really needed from the user.

The point is that for availability reason, the hypershift controller is setting a soft topologySpreadConstraint that under some circumstances (load peaks, unavailable nodes...) can lead to having two VMs of the same node pool running on the same physical machine with availability risks.
The descheduler with the `SoftTopologyAndDuplicates` profile can be used to get it periodically fixed.